File: packages/android/src/main/java/com/kozmos/components/Accordion/Accordion.kt
package com.kozmos.components.accordion

import androidx.compose.animation.AnimatedVisibility
import androidx.compose.animation.core.animateFloatAsState
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.ColumnScope
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.KeyboardArrowDown
import androidx.compose.material3.Divider
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.rotate
import com.kozmos.tokens.KozmosColors
import com.kozmos.tokens.KozmosDimensions

@Composable
fun KozmosAccordion(
    modifier: Modifier = Modifier,
    content: @Composable ColumnScope.() -> Unit
) {
    Column(modifier = modifier) {
        content()
    }
}

@Composable
fun KozmosAccordionItem(
    modifier: Modifier = Modifier,
    content: @Composable ColumnScope.() -> Unit
) {
    Column(modifier = modifier) {
        content()
    }
}

@Composable
fun KozmosAccordionTrigger(
    value: String,
    selectedValue: String?,
    onValueChange: (String?) -> Unit,
    title: String,
    modifier: Modifier = Modifier
) {
    val trackEvent = com.kozmos.providers.LocalKozmosAnalytics.current
    val isExpanded = value == selectedValue
    val rotation by animateFloatAsState(if (isExpanded) 180f else 0f, label = "rotation")

    Column(modifier = modifier) {
        Row(
            verticalAlignment = Alignment.CenterVertically,
            modifier = Modifier
                .fillMaxWidth()
                .clickable {
                    if (!isExpanded) {
                        trackEvent(com.kozmos.providers.KozmosAnalyticsEvent(component = "Accordion", eventName = "accordion_toggled", properties = mapOf("value" to value.toString())))
                        onValueChange(value)
                    } else {
                        onValueChange(null)
                    }
                }
                .padding(KozmosDimensions.primitivesLayoutSpacing200)
        ) {
            Text(
                text = title,
                modifier = Modifier.weight(1f)
            )
            Icon(
                imageVector = Icons.Default.KeyboardArrowDown,
                contentDescription = if (isExpanded) "Collapse" else "Expand",
                modifier = Modifier.rotate(rotation)
            )
        }
        Divider(color = KozmosColors.primitivesColorsBackground300)
    }
}

@Composable
fun KozmosAccordionContent(
    value: String,
    selectedValue: String?,
    modifier: Modifier = Modifier,
    content: @Composable () -> Unit
) {
    val isExpanded = value == selectedValue
    AnimatedVisibility(visible = isExpanded) {
        Column(modifier = modifier.padding(horizontal = KozmosDimensions.primitivesLayoutSpacing200, vertical = KozmosDimensions.primitivesLayoutSpacing100)) {
            content()
        }
    }
}

File: packages/ios/Sources/Components/Accordion/Accordion.swift
import SwiftUI

public struct KozmosAccordion<Content: View>: View {
    let content: Content
    public init(@ViewBuilder content: () -> Content) { self.content = content() }
    public var body: some View { VStack(spacing: 0) { content } }
}

public struct KozmosAccordionItem<Content: View>: View {
    let content: Content
    public init(@ViewBuilder content: () -> Content) { self.content = content() }
    public var body: some View { VStack(spacing: 0) { content } }
}

public struct KozmosAccordionTrigger: View {
    let value: String
    @Binding var selection: String?
    let title: String
    @Environment(\.kozmosAnalytics) var trackEvent
    
    public init(value: String, title: String, selection: Binding<String?>) {
        self.value = value
        self.title = title
        self._selection = selection
    }
    
    public var body: some View {
        Button(action: {
            let isExpanding = (selection != value)
            if isExpanding {
                 trackEvent(KozmosAnalyticsEvent(eventName: "accordion_toggled", component: "Accordion", properties: ["value": value]))
                 withAnimation { selection = value }
            } else {
                 withAnimation { selection = nil }
            }
        }) {
            HStack {
                Text(title)
                    .font(.subheadline)
                    .foregroundColor(KozmosColors.primitivesColorsForeground100)
                Spacer()
                Image(systemName: "chevron.down")
                    .rotationEffect(.degrees(selection == value ? 180 : 0))
                    .foregroundColor(KozmosColors.primitivesColorsForeground500)
            }
            .padding(.vertical, KozmosDimensions.primitivesLayoutSpacing200)
        }
        .buttonStyle(PlainButtonStyle())
        Divider().background(KozmosColors.primitivesColorsBackground300)
    }
}

public struct KozmosAccordionContent<Content: View>: View {
    let value: String
    @Binding var selection: String?
    let content: Content
    
    public init(value: String, selection: Binding<String?>, @ViewBuilder content: () -> Content) {
        self.value = value
        self._selection = selection
        self.content = content()
    }
    
    public var body: some View {
        if selection == value {
            content
                .padding(.horizontal, KozmosDimensions.primitivesLayoutSpacing200)
                .padding(.vertical, KozmosDimensions.primitivesLayoutSpacing100)
        }
    }
}
